src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.OpenFileTracker.cs

```diff
@@ -366,10 +364,8 @@ private void TryClosingDocumentsForCookie(uint cookie)
                             }
                             else
                             {
-                                // TODO: implement the ability for analyze config documents to be opened against
-                                // a live text editor. This is tracked by
-                                // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/750120
                                 Debug.Assert(w.CurrentSolution.ContainsAnalyzerConfigDocument(documentId));
```

Comment:
with textdocumentkind do we still need this multiple if statement?

Reply:
feels like we can have 1 CloseTextDocument extension method that internally use switch to find right API to close it (same for open, add, remove, change) and it shared with all code base?

Reply:
+1. I fully agree that this PR is just an intermediate step in more code sharing and shimming around these different document kinds. We would most certainly need follow-up PR(s) to rationalize the APIs here.